packages/react-query/src/__tests__/prefetch.test.tsx

```diff
@@ -0,0 +1,353 @@
+import { describe, expect, it, vi } from 'vitest'
+import React from 'react'
+import { fireEvent, waitFor } from '@testing-library/react'
+import {
+  QueryCache,
+  usePrefetchInfiniteQuery,
+  usePrefetchQuery,
+  useSuspenseInfiniteQuery,
+  useSuspenseQuery,
+} from '..'
+import { createQueryClient, queryKey, renderWithClient, sleep } from './utils'
+
+describe('usePrefetchQuery', () => {
+  const queryCache = new QueryCache()
+  const queryClient = createQueryClient({ queryCache })
+
+  it('should prefetch query if query state does not exist', async () => {
+    const queryOpts = {
+      queryKey: queryKey(),
+      queryFn: vi.fn().mockImplementation(async () => {
+        await sleep(10)
+
+        return 'Prefetch is nice!'
+      }),
+    }
+
+    const anotherSpy = vi.fn()
+
+    function Suspended() {
+      const state = useSuspenseQuery({ ...queryOpts, queryFn: anotherSpy })
+
+      return (
+        <div>
+          <div>data: {String(state.data)}</div>
+          <div>fetching: {state.isFetching ? 'true' : 'false'}</div>
+        </div>
+      )
+    }
+
+    function App() {
+      usePrefetchQuery(queryOpts)
+
+      return (
+        <React.Suspense fallback="Loading...">
+          <Suspended />
+        </React.Suspense>
+      )
+    }
+
+    const rendered = renderWithClient(queryClient, <App />)
+
+    await waitFor(() => rendered.getByText('data: Prefetch is nice!'))
+    expect(rendered.queryByText('fetching: true')).not.toBeInTheDocument()
+    expect(queryOpts.queryFn).toHaveBeenCalledTimes(1)
+    expect(anotherSpy).not.toHaveBeenCalled()
+  })
+
+  it('should not prefetch query if query state exists', async () => {
+    const queryOpts = {
+      queryKey: queryKey(),
+      queryFn: vi.fn().mockImplementation(async () => {
+        await sleep(10)
+
+        return 'Prefetch is really nice!'
+      }),
+    }
+
+    const anotherSpy = vi.fn()
+
+    function Suspended() {
+      const state = useSuspenseQuery({ ...queryOpts, queryFn: anotherSpy })
+
+      return (
+        <div>
+          <div>data: {String(state.data)}</div>
+          <div>fetching: {state.isFetching ? 'true' : 'false'}</div>
+        </div>
+      )
+    }
+
+    function App() {
+      usePrefetchQuery({ ...queryOpts, queryFn: anotherSpy })
+
+      return (
+        <React.Suspense fallback="Loading...">
+          <Suspended />
+        </React.Suspense>
+      )
+    }
+
+    await queryClient.fetchQuery(queryOpts)
+    const rendered = renderWithClient(queryClient, <App />)
+
+    expect(rendered.queryByText('fetching: true')).not.toBeInTheDocument()
+    expect(
+      rendered.getByText('data: Prefetch is really nice!'),
+    ).toBeInTheDocument()
+    expect(anotherSpy).not.toHaveBeenCalled()
```

Comment:
let's also add:

```suggestion
    expect(anotherSpy).not.toHaveBeenCalled()
    expect(queryOpts.queryFn).not.toHaveBeenCalled()
```

Reply:
I think it's better to call `mockClear()` on the spy after you've called `fetchQuery`

Reply:
Yep, good idea! Updated :smile: